client/my-sites/site-settings/main.jsx

```diff
@@ -90,7 +78,7 @@ export class SiteSettingsComponent extends Component {
 	}
 
 	render() {
-		const { site } = this.state;
+		const { site } = this.props;
```

Comment:
`site` is passed as a connected prop for `<SiteSettingsNavigation />`, so it's no longer necessary to be specifically passed to it, you can remove it.

Also, I think after you remove it from here, `site` will no longer need to be added as a connected prop, so it can be removed from there, too. You can use `siteId` for the checks in `render()`.